src/package.h

```diff
@@ -9,6 +9,7 @@
 
 #include "error.h"
 #include "logger.h"
+#include "random_number_generator.h"
```

Comment:
I wonder if we can just forward declare the classes here rather than importing the full header?
```suggestion
class DoubleDistribution;
class DoubleDistribution::Ptr;
```
If this works, we'll need to include the header in the `package.cc` file instead.

If it works, it's preferred because it reduced the dependency graph. If it stays this way, every file that includes `package.h` will include the `random_number_generator.h` and all of the things it includes.  Any change in any file down that dependency tree causes the whole stack to be rebuilt.

If any part of the interface is uses in the package header, then it would need to be included.  But if we only need to declare that the classes exist, then this can work instead.  I'm just not sure about the `::Ptr` part..

Reply:
I'm still getting `invalid use of incomplete type` errors on the `::Ptr`s even on my machine after including the RNG in `package.cc`... I think this means I need to include it in the header